src/download.c

```diff
@@ -452,6 +452,9 @@ static int perform_curl_io_and_complete(int count, bool bounded)
 	return 0;
 }
 
+/* Fixed window added to hysteresis upper bound before it is enforced. */
+#define BUFFER 5
```

Comment:
Alternatively, we could decide to change the hysteresis upper/lower bounds entirely instead of using a "buffer" like this. But I would want to do a bunch of experimentation first before considering that.